added chatState in the globalThis, added navmesh to ground plane and …

alextitonis opened the pull request:
…made an override system for the movement commands

## Summary

A summary of changes being made in this PR

## Checklist
- [ ] Pre-push checks pass `npm run check`
  - [ ] Linter passing via `npm run lint`
  - [ ] Unit & Integration tests passing via `npm run test:packages`
  - [ ] Docker build process passing via `npm run build-client`
- [ ] Changes have been manually QA'd 
- [ ] Changes reviewed by at least 2 approved reviewers

## References

References to pertaining issue(s)

## Reviewers

@HexaField @speigg @NateTheGreatt

patreeceeo commented:
Hey Alex, would you mind adding some explanation in the PR description?

speigg commented on packages/engine/src/networking/utils/getUser.ts:
```diff
@@ -7,11 +7,21 @@ import { UserNameComponent } from '../../scene/components/UserNameComponent'
   })
-  return client ? useWorld().getUserAvatarEntity(client.userId) : undefined
+  return client ? useWorld().getUserAvatarEntity(client.userId) : undefined*/
+  const world = useWorld()
```
If this function is meant to be called outside of a System, then it's preferable to use Engine.defaultWorld rather than `useWorld`

alextitonis replied:
updated it

HexaField commented on packages/client-core/src/social/services/ChatService.ts:
```diff
@@ -215,6 +215,8 @@ export const accessChatState = () => state
 
 export const useChatState = () => useState(state) as any as typeof state
 
+globalThis.chatState = state
```
why does chat state need to be on the global state? can we use the accessChatState hook instead?

HexaField commented on packages/engine/src/scene/functions/createGround.ts:
```diff
@@ -33,5 +37,15 @@ export const createGround = async function (entity: Entity, args: GroundProps, i
 
   createCollider(entity, mesh)
 
+  if (isClient) {
```
lets move this to the navigation system and add a GroundPlaneTagComponent component to query for, to keep this decouple from the core engine functionality

HexaField replied:
For now, maybe we can just have an option in the editor on the Scene node for if the scene should have navmeshes or not. That way we can optimise scenes that don't need it, and have a single source of truth per-secne.

alextitonis commented:
> 
> 
> Hey Alex, would you mind adding some explanation in the PR description?

In the PR, there are 2 things done, first i've added the chat state in the globalThis, so i can access it through puppeteer.evaluate
And added navmesh based on the Map object in the ground plane and added an override for the /move /follow commands, to set exact target position ignoring the raycasting (x,y,z)

shawwwwwwww commented:
lgtm

patreeceeo commented:
Can we get rid these console logs if they aren't used anymore? And any unused args/imports/declarations

HexaField commented on packages/editor/src/nodes/GroundPlaneNode.ts:
```diff
@@ -70,5 +72,8 @@ export default class GroundPlaneNode extends EditorNodeMixin(GroundPlane) {
       receive: this.receiveShadow,
       cast: false
     })
+    this.addGLTFComponent('navmesh', {
```
this is not needed, generateNavmesh will be serialised into `ground-plane` and `navmesh` has no deserialization in either the location nor the editor
